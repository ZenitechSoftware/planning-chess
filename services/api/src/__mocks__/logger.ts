const mockLogger = {
  info: jest.fn(),
  error: jest.fn(),
  debug: jest.fn(),
};

export default mockLogger;
