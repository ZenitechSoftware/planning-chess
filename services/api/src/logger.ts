import winston from 'winston';

const logger = winston.createLogger({
  transports: [
    new winston.transports.Console({
      level: process.env.NODE_ENV === 'test' ? 'crit' : 'debug',
      format: winston.format.combine(
        winston.format.colorize(),
        winston.format.simple(),
      ),
    }),
  ],
});

export default logger;
