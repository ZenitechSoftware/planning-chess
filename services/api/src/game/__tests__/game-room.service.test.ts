import * as gameRoomService from '../game-room.service';
import logger from '../../logger';

jest.mock('uuid', () => ({
  v4: () => 'some-short-v4-uuid-0',
}));

jest.mock('../../logger');

describe('gameRoomService', () => {
  afterEach(() => {
    jest.clearAllMocks();
    gameRoomService.rooms.clear();
  });

  const roomId = 'abcd-1234';

  it('should create new room', async () => {
    const room = gameRoomService.getOrCreateRoom();
    expect(room).toMatchSnapshot();
  });

  it('should get existing room', async () => {
    gameRoomService.getOrCreateRoom(roomId);
    const secondCall = gameRoomService.getOrCreateRoom(roomId);
    expect(secondCall).toMatchSnapshot();
  });

  it('should print rooms', async () => {
    gameRoomService.getOrCreateRoom(roomId);
    gameRoomService.printRooms();
    expect(logger.debug).toBeCalled();
  });

  it('should not clean up room, because there are less than 1000', () => {
    gameRoomService.getOrCreateRoom(roomId);
    gameRoomService.cleanUp();
    expect(gameRoomService.getClients(roomId)).toBeDefined();
  });

  it('should clean up the room, because there are more than 1000', () => {
    for (let i = 0; i < 1001; i++) {
      gameRoomService.getOrCreateRoom(`${i}`);
    }
    expect(gameRoomService.rooms.size).toBe(1001);
    gameRoomService.cleanUp();
    expect(gameRoomService.rooms.size).toBe(1000);
  });
});
