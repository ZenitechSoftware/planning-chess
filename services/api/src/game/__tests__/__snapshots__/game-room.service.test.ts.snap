// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gameRoomService should create new room 1`] = `
Object {
  "gameState": "GAME_NOT_STARTED",
  "players": Map {},
  "roomId": "some-short-v4-uuid-0",
  "server": WebSocketServer {
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_shouldEmitClose": false,
    "_state": 0,
    "clients": Set {},
    "options": Object {
      "WebSocket": [Function],
      "backlog": null,
      "clientTracking": true,
      "handleProtocols": null,
      "host": null,
      "maxPayload": 104857600,
      "noServer": true,
      "path": null,
      "perMessageDeflate": false,
      "port": null,
      "server": null,
      "skipUTF8Validation": false,
      "verifyClient": null,
    },
    Symbol(kCapture): false,
  },
  "turns": Array [],
}
`;

exports[`gameRoomService should get existing room 1`] = `
Object {
  "gameState": "GAME_NOT_STARTED",
  "players": Map {},
  "roomId": "abcd-1234",
  "server": WebSocketServer {
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_shouldEmitClose": false,
    "_state": 0,
    "clients": Set {},
    "options": Object {
      "WebSocket": [Function],
      "backlog": null,
      "clientTracking": true,
      "handleProtocols": null,
      "host": null,
      "maxPayload": 104857600,
      "noServer": true,
      "path": null,
      "perMessageDeflate": false,
      "port": null,
      "server": null,
      "skipUTF8Validation": false,
      "verifyClient": null,
    },
    Symbol(kCapture): false,
  },
  "turns": Array [],
}
`;
